app/messaging/process-vet-visit.js

```diff
@@ -0,0 +1,37 @@
+const util = require('util')
+const { set } = require('../repositories/vet-visit-repository')
+const { get } = require('../repositories/application-repository')
+const sendMessage = require('../messaging/send-message')
+const { vetVisitResponseMsgType, applicationResponseQueue } = require('../config')
+const sendEmail = require('../lib/send-email')
+const { validateApplication } = require('./validate-message')
+
+const processVetVisit = async (message) => {
+  try {
+    const msgBody = message.body
+    console.log('received process vet visit request', util.inspect(msgBody, false, null, true))
+    const { reference } = msgBody.signup
+    const farmerApplication = await get(reference)
+
+    if (validateApplication(farmerApplication) === false) {
+      return sendMessage({ applicationState: 'failed' }, vetVisitResponseMsgType, applicationResponseQueue, { sessionId: msgBody.sessionId })
```

Comment:
I was thinking there would be a state for when no application exists and another for when it had been submitted already. Then the message in the frontend could be specific to those scenarios. What do you think?